From c5fdeb3072ad3558db7918c9da09ffc529796aca Mon Sep 17 00:00:00 2001
From: Shivam Mathur <shivam_jpr@hotmail.com>
Date: Sat, 20 Jul 2024 22:05:11 +0000
Subject: [PATCH] [PATCH] ext/intl: level up c++ runtime std for icu 74 and
 onwards.

---
 ext/intl/config.m4 | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/ext/intl/config.m4 b/ext/intl/config.m4
index 4b9f469b38..7b5a9dd658 100644
--- a/ext/intl/config.m4
+++ b/ext/intl/config.m4
@@ -89,6 +89,16 @@ if test "$PHP_INTL" != "no"; then
     breakiterator/rulebasedbreakiterator_methods.cpp \
     breakiterator/codepointiterator_internal.cpp \
     breakiterator/codepointiterator_methods.cpp"
+
+  AC_MSG_CHECKING([if intl requires -std=gnu++17])
+  AS_IF([test "$PKG_CONFIG icu-uc --atleast-version=74"],[
+    AC_MSG_RESULT([yes])
+    PHP_CXX_COMPILE_STDCXX(17, mandatory, PHP_INTL_STDCXX)
+  ],[
+    AC_MSG_RESULT([no])
+    PHP_CXX_COMPILE_STDCXX(11, mandatory, PHP_INTL_STDCXX)
+  ])
+
   PHP_INTL_CXX_FLAGS="$INTL_COMMON_FLAGS $ICU_CXXFLAGS"
   if test "$ext_shared" = "no"; then
     PHP_ADD_SOURCES(PHP_EXT_DIR(intl), $PHP_INTL_CXX_SOURCES, $PHP_INTL_CXX_FLAGS)
-- 
2.45.2

