From 2cb3275372aa59f88b934e8c8b606830e62d5380 Mon Sep 17 00:00:00 2001
From: Shivam Mathur <shivam_jpr@hotmail.com>
Date: Sat, 20 Jul 2024 22:14:43 +0000
Subject: [PATCH] [PATCH] ext/intl: level up c++ runtime std for icu 74 and
 onwards.

---
 ext/intl/config.m4 | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/ext/intl/config.m4 b/ext/intl/config.m4
index 4630a302ef..872ad1c87e 100644
--- a/ext/intl/config.m4
+++ b/ext/intl/config.m4
@@ -86,6 +86,16 @@ if test "$PHP_INTL" != "no"; then
     breakiterator/codepointiterator_methods.cpp \
     idn/idn.c \
     $icu_spoof_src, $ext_shared,,$ICU_INCS -Wno-write-strings)
+
+  AC_MSG_CHECKING([if intl requires -std=gnu++17])
+  AS_IF([test "$PKG_CONFIG icu-uc --atleast-version=74"],[
+    AC_MSG_RESULT([yes])
+    PHP_CXX_COMPILE_STDCXX(17, mandatory, PHP_INTL_STDCXX)
+  ],[
+    AC_MSG_RESULT([no])
+    PHP_CXX_COMPILE_STDCXX(11, mandatory, PHP_INTL_STDCXX)
+  ])
+
   PHP_ADD_BUILD_DIR($ext_builddir/collator)
   PHP_ADD_BUILD_DIR($ext_builddir/converter)
   PHP_ADD_BUILD_DIR($ext_builddir/common)
-- 
2.45.2

